examples_gbs/run_tutorial_dense.py

```diff
@@ -0,0 +1,120 @@
+# pylint: disable=wrong-import-position,wrong-import-order,ungrouped-imports
+"""
+Dense Subgraph Tutorial
+=========================
+Graphs can be used to model a wide variety of objects: social networks, financial markets,
+biological networks, and many others. A common problem of interest is to find subgraphs that
+contain a large number of connections between its nodes. These may correspond to communities in
+social networks, correlated assets in a market, or mutually influential proteins in a biological
+network. This tutorial studies how GBS can be used to identify dense subgraphs.
+
+Mathematically, this task is known as the `dense subgraph problem
+<https://en.wikipedia.org/wiki/Dense_subgraph>`_. The density of a graph is equal to the number of
+edges in the subgraph divided by the maximum possible number of edges. Identifying the densest
+graph of a given size, known as the densest-*k* subgraph problem, is `NP-Hard
+<https://en.wikipedia.org/wiki/NP-hardness>`_.
+
+
+As shown in :cite:`arrazola2018using`, a defining feature of GBS is that when we encode a graph
+into a GBS device, it samples dense subgraphs with high probability. This property can be
+used to find dense subgraphs by sampling from a GBS device  and postprocessing the outputs.
+Let's take a look!
+
+Finding dense subgraphs
+-----------------------
+As usual, the first step is to import all required modules. We'll need the :mod:`~.gbs.data`
+module to load pre-generated samples, the :mod:`~.gbs.sample` module to postselect samples, the 
+:mod:`~.gbs.subgraph` module to search for dense subgraphs, and the :mod:`~.gbs.plot` module to 
+visualize the graphs. We'll also use Plotly for plotting and NetworkX for graph

```

Comment:
> We'll also use Plotly for plotting

Does it raise the question of why import the `plot` module?

Reply:
Changed to "We'll also use Plotly which is required for the :mod:`~.gbs.plot` module"

Reply:
Actually maybe we should not mention the import of `plotly` at all, since the user shouldn't need it in general _unless_ they want to use `plotly.offline.plot()`, which we say shouldn't typically be the case.